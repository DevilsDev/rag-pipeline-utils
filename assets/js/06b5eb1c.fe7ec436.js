"use strict";(self.webpackChunk_devilsdev_rag_pipeline_utils=self.webpackChunk_devilsdev_rag_pipeline_utils||[]).push([[1439],{148:e=>{e.exports=JSON.parse('{"permalink":"/rag-pipeline-utils/blog/image-showcase","editUrl":"https://github.com/DevilsDev/rag-pipeline-utils/edit/main/docs-site/blog/blog/2025-04-20-image-showcase.md","source":"@site/blog/2025-04-20-image-showcase.md","title":"Image Showcase with LQIP","description":"Demonstration of how low-quality image placeholders (LQIP) improve blog UX and optimize web performance.","date":"2025-04-20T00:00:00.000Z","tags":[{"inline":false,"label":"Image","permalink":"/rag-pipeline-utils/blog/tags/blog/tags/image"},{"inline":false,"label":"Tutorials","permalink":"/rag-pipeline-utils/blog/tags/blog/tags/tutorials"}],"readingTime":0.165,"hasTruncateMarker":true,"authors":[{"name":"Ali Kahwaji","title":"Creator & Maintainer of RAG Pipeline Utils","url":"https://github.com/alikahwaji","imageURL":"/rag-pipeline-utils/img/authors/ali.jpg","key":"ali","page":null}],"frontMatter":{"slug":"image-showcase","title":"Image Showcase with LQIP","authors":["ali"],"tags":["image","tutorials"]},"unlisted":false,"prevItem":{"title":"\u2728 Version v2.1.3 Released","permalink":"/rag-pipeline-utils/blog/release-v2.1.3"},"nextItem":{"title":"release-2.1.1","permalink":"/rag-pipeline-utils/blog/2025/04/20/release-2.1.1"}}')},2817:(e,t,i)=>{i.r(t),i.d(t,{assets:()=>r,contentTitle:()=>n,default:()=>g,frontMatter:()=>o,metadata:()=>a,toc:()=>c});var a=i(148),l=i(6394),s=i(3755);const o={slug:"image-showcase",title:"Image Showcase with LQIP",authors:["ali"],tags:["image","tutorials"]},n=void 0,r={authorsImageUrls:[void 0]},c=[{value:"Details",id:"details",level:2}];function p(e){const t={code:"code",h2:"h2",p:"p",...(0,s.R)(),...e.components};return(0,l.jsxs)(l.Fragment,{children:[(0,l.jsx)(t.p,{children:"Demonstration of how low-quality image placeholders (LQIP) improve blog UX and optimize web performance."}),"\n",(0,l.jsx)(t.h2,{id:"details",children:"Details"}),"\n",(0,l.jsxs)(t.p,{children:["This post showcases ",(0,l.jsx)(t.code,{children:"BlogImage"})," and ",(0,l.jsx)(t.code,{children:"BlogCard"})," with LQIP integration via Plaiceholder and SSR-safe fallbacks."]})]})}function g(e={}){const{wrapper:t}={...(0,s.R)(),...e.components};return t?(0,l.jsx)(t,{...e,children:(0,l.jsx)(p,{...e})}):p(e)}},3755:(e,t,i)=>{i.d(t,{R:()=>o,x:()=>n});var a=i(2610);const l={},s=a.createContext(l);function o(e){const t=a.useContext(s);return a.useMemo((function(){return"function"==typeof e?e(t):{...t,...e}}),[t,e])}function n(e){let t;return t=e.disableParentContext?"function"==typeof e.components?e.components(l):e.components||l:o(e.components),a.createElement(s.Provider,{value:t},e.children)}}}]);