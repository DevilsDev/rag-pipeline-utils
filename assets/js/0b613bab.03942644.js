"use strict";(self.webpackChunk_devilsdev_rag_pipeline_utils=self.webpackChunk_devilsdev_rag_pipeline_utils||[]).push([[8880],{1369:e=>{e.exports=JSON.parse('{"permalink":"/rag-pipeline-utils/blog/prompt-engineering-best-practices","editUrl":"https://github.com/DevilsDev/rag-pipeline-utils/edit/main/docs-site/blog/blog/2025-04-19-prompt-engineering-best-practices.md","source":"@site/blog/2025-04-19-prompt-engineering-best-practices.md","title":"Prompt Engineering Best Practices","description":"Design better prompts for your LLM-driven RAG workflows using templating, in-context examples, and task-specific embeddings.","date":"2025-04-19T00:00:00.000Z","tags":[{"inline":false,"label":"Prompt Engineering","permalink":"/rag-pipeline-utils/blog/tags/blog/tags/prompt-engineering"},{"inline":false,"label":"Tutorials","permalink":"/rag-pipeline-utils/blog/tags/blog/tags/tutorials"}],"readingTime":0.18,"hasTruncateMarker":true,"authors":[{"name":"Ali Kahwaji","title":"Creator & Maintainer of RAG Pipeline Utils","url":"https://github.com/alikahwaji","imageURL":"/rag-pipeline-utils/img/authors/ali.jpg","key":"ali","page":null}],"frontMatter":{"slug":"prompt-engineering-best-practices","title":"Prompt Engineering Best Practices","authors":["ali"],"tags":["prompt-engineering","tutorials"]},"unlisted":false,"prevItem":{"title":"release-2.1.1","permalink":"/rag-pipeline-utils/blog/2025/04/20/release-2.1.1"},"nextItem":{"title":"RAG Architecture Overview","permalink":"/rag-pipeline-utils/blog/architecture-overview"}}')},3755:(e,t,i)=>{i.d(t,{R:()=>l,x:()=>a});var n=i(2610);const r={},s=n.createContext(r);function l(e){const t=n.useContext(s);return n.useMemo((function(){return"function"==typeof e?e(t):{...t,...e}}),[t,e])}function a(e){let t;return t=e.disableParentContext?"function"==typeof e.components?e.components(r):e.components||r:l(e.components),n.createElement(s.Provider,{value:t},e.children)}},8960:(e,t,i)=>{i.r(t),i.d(t,{assets:()=>o,contentTitle:()=>a,default:()=>c,frontMatter:()=>l,metadata:()=>n,toc:()=>p});var n=i(1369),r=i(6394),s=i(3755);const l={slug:"prompt-engineering-best-practices",title:"Prompt Engineering Best Practices",authors:["ali"],tags:["prompt-engineering","tutorials"]},a=void 0,o={authorsImageUrls:[void 0]},p=[{value:"Tips",id:"tips",level:2}];function g(e){const t={h2:"h2",li:"li",p:"p",ul:"ul",...(0,s.R)(),...e.components};return(0,r.jsxs)(r.Fragment,{children:[(0,r.jsx)(t.p,{children:"Design better prompts for your LLM-driven RAG workflows using templating, in-context examples, and task-specific embeddings."}),"\n",(0,r.jsx)(t.h2,{id:"tips",children:"Tips"}),"\n",(0,r.jsxs)(t.ul,{children:["\n",(0,r.jsx)(t.li,{children:"Use instructions + context"}),"\n",(0,r.jsx)(t.li,{children:"Dynamically inject retrieved passages"}),"\n",(0,r.jsx)(t.li,{children:"Validate through CLI or dashboard"}),"\n"]})]})}function c(e={}){const{wrapper:t}={...(0,s.R)(),...e.components};return t?(0,r.jsx)(t,{...e,children:(0,r.jsx)(g,{...e})}):g(e)}}}]);