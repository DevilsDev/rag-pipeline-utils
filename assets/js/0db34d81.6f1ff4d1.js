"use strict";(self.webpackChunk_devilsdev_rag_pipeline_utils=self.webpackChunk_devilsdev_rag_pipeline_utils||[]).push([[5499],{1362:e=>{e.exports=JSON.parse('{"archive":{"blogPosts":[{"id":"release-v2.1.6","metadata":{"permalink":"/rag-pipeline-utils/blog/release-v2.1.6","editUrl":"https://github.com/DevilsDev/rag-pipeline-utils/edit/main/docs-site/blog/blog/2025-04-24-v2.1.6.md","source":"@site/blog/2025-04-24-v2.1.6.md","title":"\ud83d\ude80 Version v2.1.6 Released","description":"RAG Pipeline Utils v2.1.6 is now available on NPM!","date":"2025-04-24T00:00:00.000Z","tags":[{"inline":false,"label":"Release","permalink":"/rag-pipeline-utils/blog/tags/blog/tags/release"},{"inline":false,"label":"Changelog","permalink":"/rag-pipeline-utils/blog/tags/blog/tags/changelog"}],"readingTime":0.305,"hasTruncateMarker":false,"authors":[{"name":"Ali Kahwaji","title":"Creator & Maintainer of RAG Pipeline Utils","url":"https://github.com/alikahwaji","imageURL":"/rag-pipeline-utils/img/authors/ali.jpg","key":"ali","page":null}],"frontMatter":{"slug":"release-v2.1.6","title":"\ud83d\ude80 Version v2.1.6 Released","authors":["ali"],"tags":["release","changelog"]},"unlisted":false,"nextItem":{"title":"\u2728 Version v2.1.5 Released","permalink":"/rag-pipeline-utils/blog/release-v2.1.5"}},"content":"RAG Pipeline Utils **v2.1.6** is now available on NPM!\\n\\n## \ud83d\udd27 Changes\\n\\n- chore: remove legacy changelog workflow and finalize blog badge\\n- ci(workflows): add blog badge verification workflow\\n- ci(workflows): add blog badge verification workflow\\n- ci(workflows): add blog badge verification workflow\\n- docs(blog): add release post for v2.1.5\\n\\n## \ud83d\udd17 Resources\\n\\n- GitHub Compare: [v2.1.5...v2.1.6](https://github.com/DevilsDev/rag-pipeline-utils/compare/v2.1.5...v2.1.6)\\n- View full [CHANGELOG.md](../../CHANGELOG.md)"},{"id":"release-v2.1.5","metadata":{"permalink":"/rag-pipeline-utils/blog/release-v2.1.5","editUrl":"https://github.com/DevilsDev/rag-pipeline-utils/edit/main/docs-site/blog/blog/2025-04-22-release-v2.1.5.md","source":"@site/blog/2025-04-22-release-v2.1.5.md","title":"\u2728 Version v2.1.5 Released","description":"RAG Pipeline Utils vv2.1.5 is now available on NPM!","date":"2025-04-22T00:00:00.000Z","tags":[{"inline":false,"label":"Release","permalink":"/rag-pipeline-utils/blog/tags/blog/tags/release"},{"inline":false,"label":"Changelog","permalink":"/rag-pipeline-utils/blog/tags/blog/tags/changelog"}],"readingTime":0.15,"hasTruncateMarker":false,"authors":[{"name":"Ali Kahwaji","title":"Creator & Maintainer of RAG Pipeline Utils","url":"https://github.com/alikahwaji","imageURL":"/rag-pipeline-utils/img/authors/ali.jpg","key":"ali","page":null}],"frontMatter":{"slug":"release-v2.1.5","title":"\u2728 Version v2.1.5 Released","authors":["ali"],"tags":["release","changelog"]},"unlisted":false,"prevItem":{"title":"\ud83d\ude80 Version v2.1.6 Released","permalink":"/rag-pipeline-utils/blog/release-v2.1.6"},"nextItem":{"title":"\u2728 Version v2.1.3 Released","permalink":"/rag-pipeline-utils/blog/release-v2.1.3"}},"content":"RAG Pipeline Utils vv2.1.5 is now available on NPM!\\n\\n#### Highlights\\n- \ud83d\ude80 Auto-published blog from GitHub Release\\n- \ud83d\udce6 Version bump: v2.1.5\\n- \ud83d\udcd8 See CHANGELOG.md for full details"},{"id":"release-v2.1.3","metadata":{"permalink":"/rag-pipeline-utils/blog/release-v2.1.3","editUrl":"https://github.com/DevilsDev/rag-pipeline-utils/edit/main/docs-site/blog/blog/2025-04-21-release-v2.1.3.md","source":"@site/blog/2025-04-21-release-v2.1.3.md","title":"\u2728 Version v2.1.3 Released","description":"RAG Pipeline Utils vv2.1.3 is now available on NPM!","date":"2025-04-21T00:00:00.000Z","tags":[{"inline":false,"label":"Release","permalink":"/rag-pipeline-utils/blog/tags/blog/tags/release"},{"inline":false,"label":"Changelog","permalink":"/rag-pipeline-utils/blog/tags/blog/tags/changelog"}],"readingTime":0.15,"hasTruncateMarker":false,"authors":[{"name":"Ali Kahwaji","title":"Creator & Maintainer of RAG Pipeline Utils","url":"https://github.com/alikahwaji","imageURL":"/rag-pipeline-utils/img/authors/ali.jpg","key":"ali","page":null}],"frontMatter":{"slug":"release-v2.1.3","title":"\u2728 Version v2.1.3 Released","authors":["ali"],"tags":["release","changelog"]},"unlisted":false,"prevItem":{"title":"\u2728 Version v2.1.5 Released","permalink":"/rag-pipeline-utils/blog/release-v2.1.5"},"nextItem":{"title":"\u2728 Version v2.1.4 Released","permalink":"/rag-pipeline-utils/blog/release-v2.1.4"}},"content":"RAG Pipeline Utils vv2.1.3 is now available on NPM!\\n\\n#### Highlights\\n- \ud83d\ude80 Auto-published blog from GitHub Release\\n- \ud83d\udce6 Version bump: v2.1.3\\n- \ud83d\udcd8 See CHANGELOG.md for full details"},{"id":"release-v2.1.4","metadata":{"permalink":"/rag-pipeline-utils/blog/release-v2.1.4","editUrl":"https://github.com/DevilsDev/rag-pipeline-utils/edit/main/docs-site/blog/blog/2025-04-21-release-v2.1.4.md","source":"@site/blog/2025-04-21-release-v2.1.4.md","title":"\u2728 Version v2.1.4 Released","description":"RAG Pipeline Utils vv2.1.4 is now available on NPM!","date":"2025-04-21T00:00:00.000Z","tags":[{"inline":false,"label":"Release","permalink":"/rag-pipeline-utils/blog/tags/blog/tags/release"},{"inline":false,"label":"Changelog","permalink":"/rag-pipeline-utils/blog/tags/blog/tags/changelog"}],"readingTime":0.15,"hasTruncateMarker":false,"authors":[{"name":"Ali Kahwaji","title":"Creator & Maintainer of RAG Pipeline Utils","url":"https://github.com/alikahwaji","imageURL":"/rag-pipeline-utils/img/authors/ali.jpg","key":"ali","page":null}],"frontMatter":{"slug":"release-v2.1.4","title":"\u2728 Version v2.1.4 Released","authors":["ali"],"tags":["release","changelog"]},"unlisted":false,"prevItem":{"title":"\u2728 Version v2.1.3 Released","permalink":"/rag-pipeline-utils/blog/release-v2.1.3"},"nextItem":{"title":"Image Showcase with LQIP","permalink":"/rag-pipeline-utils/blog/image-showcase"}},"content":"RAG Pipeline Utils vv2.1.4 is now available on NPM!\\n\\n#### Highlights\\n- \ud83d\ude80 Auto-published blog from GitHub Release\\n- \ud83d\udce6 Version bump: v2.1.4\\n- \ud83d\udcd8 See CHANGELOG.md for full details"},{"id":"image-showcase","metadata":{"permalink":"/rag-pipeline-utils/blog/image-showcase","editUrl":"https://github.com/DevilsDev/rag-pipeline-utils/edit/main/docs-site/blog/blog/2025-04-20-image-showcase.md","source":"@site/blog/2025-04-20-image-showcase.md","title":"Image Showcase with LQIP","description":"Demonstration of how low-quality image placeholders (LQIP) improve blog UX and optimize web performance.","date":"2025-04-20T00:00:00.000Z","tags":[{"inline":false,"label":"Image","permalink":"/rag-pipeline-utils/blog/tags/blog/tags/image"},{"inline":false,"label":"Tutorials","permalink":"/rag-pipeline-utils/blog/tags/blog/tags/tutorials"}],"readingTime":0.165,"hasTruncateMarker":true,"authors":[{"name":"Ali Kahwaji","title":"Creator & Maintainer of RAG Pipeline Utils","url":"https://github.com/alikahwaji","imageURL":"/rag-pipeline-utils/img/authors/ali.jpg","key":"ali","page":null}],"frontMatter":{"slug":"image-showcase","title":"Image Showcase with LQIP","authors":["ali"],"tags":["image","tutorials"]},"unlisted":false,"prevItem":{"title":"\u2728 Version v2.1.4 Released","permalink":"/rag-pipeline-utils/blog/release-v2.1.4"},"nextItem":{"title":"release-2.1.1","permalink":"/rag-pipeline-utils/blog/2025/04/20/release-2.1.1"}},"content":"Demonstration of how low-quality image placeholders (LQIP) improve blog UX and optimize web performance.\\n\\n\x3c!-- truncate --\x3e\\n\\n## Details\\n\\nThis post showcases `BlogImage` and `BlogCard` with LQIP integration via Plaiceholder and SSR-safe fallbacks."},{"id":"/2025/04/20/release-2.1.1","metadata":{"permalink":"/rag-pipeline-utils/blog/2025/04/20/release-2.1.1","editUrl":"https://github.com/DevilsDev/rag-pipeline-utils/edit/main/docs-site/blog/blog/2025-04-20-release-2.1.1.md","source":"@site/blog/2025-04-20-release-2.1.1.md","title":"release-2.1.1","description":"Published: 2025-04-20","date":"2025-04-20T00:00:00.000Z","tags":[],"readingTime":0.19,"hasTruncateMarker":false,"authors":[],"frontMatter":{},"unlisted":false,"prevItem":{"title":"Image Showcase with LQIP","permalink":"/rag-pipeline-utils/blog/image-showcase"},"nextItem":{"title":"Prompt Engineering Best Practices","permalink":"/rag-pipeline-utils/blog/prompt-engineering-best-practices"}},"content":"\x3c!-- .github/release-template.md --\x3e\\n\\n# \ud83d\udd16 Release v2.1.1\\n\\n**Published:** `2025-04-20`  \\n**Compare changes:** [View diff](https://github.com/DevilsDev/rag-pipeline-utils/compare/v2.0.0...v2.1.1)  \\n**Contributors:** `3`  \\n**Commits:** `17`\\n\\n---\\n\\n## \ud83d\udce6 What\'s New\\n\\n- Placeholder for changelog details or highlights.\\n\\n---\\n\\n## \ud83d\udcd8 Changelog\\n\\nSee full details in [CHANGELOG.md](/CHANGELOG.md)"},{"id":"prompt-engineering-best-practices","metadata":{"permalink":"/rag-pipeline-utils/blog/prompt-engineering-best-practices","editUrl":"https://github.com/DevilsDev/rag-pipeline-utils/edit/main/docs-site/blog/blog/2025-04-19-prompt-engineering-best-practices.md","source":"@site/blog/2025-04-19-prompt-engineering-best-practices.md","title":"Prompt Engineering Best Practices","description":"Design better prompts for your LLM-driven RAG workflows using templating, in-context examples, and task-specific embeddings.","date":"2025-04-19T00:00:00.000Z","tags":[{"inline":false,"label":"Prompt Engineering","permalink":"/rag-pipeline-utils/blog/tags/blog/tags/prompt-engineering"},{"inline":false,"label":"Tutorials","permalink":"/rag-pipeline-utils/blog/tags/blog/tags/tutorials"}],"readingTime":0.18,"hasTruncateMarker":true,"authors":[{"name":"Ali Kahwaji","title":"Creator & Maintainer of RAG Pipeline Utils","url":"https://github.com/alikahwaji","imageURL":"/rag-pipeline-utils/img/authors/ali.jpg","key":"ali","page":null}],"frontMatter":{"slug":"prompt-engineering-best-practices","title":"Prompt Engineering Best Practices","authors":["ali"],"tags":["prompt-engineering","tutorials"]},"unlisted":false,"prevItem":{"title":"release-2.1.1","permalink":"/rag-pipeline-utils/blog/2025/04/20/release-2.1.1"},"nextItem":{"title":"RAG Architecture Overview","permalink":"/rag-pipeline-utils/blog/architecture-overview"}},"content":"Design better prompts for your LLM-driven RAG workflows using templating, in-context examples, and task-specific embeddings.\\n\\n\x3c!-- truncate --\x3e\\n\\n## Tips\\n\\n- Use instructions + context\\n- Dynamically inject retrieved passages\\n- Validate through CLI or dashboard"},{"id":"architecture-overview","metadata":{"permalink":"/rag-pipeline-utils/blog/architecture-overview","editUrl":"https://github.com/DevilsDev/rag-pipeline-utils/edit/main/docs-site/blog/blog/2025-04-18-architecture-overview.md","source":"@site/blog/2025-04-18-architecture-overview.md","title":"RAG Architecture Overview","description":"Explore the modular RAG pipeline architecture. Components like loaders, embedders, retrievers, and generators are pluggable and testable.","date":"2025-04-18T00:00:00.000Z","tags":[{"inline":false,"label":"Architecture","permalink":"/rag-pipeline-utils/blog/tags/blog/tags/architecture"},{"inline":false,"label":"Modular","permalink":"/rag-pipeline-utils/blog/tags/blog/tags/modular"}],"readingTime":0.22,"hasTruncateMarker":true,"authors":[{"name":"Ali Kahwaji","title":"Creator & Maintainer of RAG Pipeline Utils","url":"https://github.com/alikahwaji","imageURL":"/rag-pipeline-utils/img/authors/ali.jpg","key":"ali","page":null}],"frontMatter":{"slug":"architecture-overview","title":"RAG Architecture Overview","authors":["ali"],"tags":["architecture","modular"]},"unlisted":false,"prevItem":{"title":"Prompt Engineering Best Practices","permalink":"/rag-pipeline-utils/blog/prompt-engineering-best-practices"},"nextItem":{"title":"\u2728 Version v2.1.1 Released","permalink":"/rag-pipeline-utils/blog/release-v2.1.1"}},"content":"Explore the modular RAG pipeline architecture. Components like loaders, embedders, retrievers, and generators are pluggable and testable.\\n\\n\x3c!-- truncate --\x3e\\n\\n## Key Concepts\\n\\n- Plugin registry for clean composition\\n- Evaluation layer for metrics like BLEU and recall\\n- CLI integration for dev-first workflows"},{"id":"release-v2.1.1","metadata":{"permalink":"/rag-pipeline-utils/blog/release-v2.1.1","editUrl":"https://github.com/DevilsDev/rag-pipeline-utils/edit/main/docs-site/blog/blog/2025-04-18-release-v2.1.1.md","source":"@site/blog/2025-04-18-release-v2.1.1.md","title":"\u2728 Version v2.1.1 Released","description":"RAG Pipeline Utils vv2.1.1 is now available on NPM!","date":"2025-04-18T00:00:00.000Z","tags":[{"inline":false,"label":"Release","permalink":"/rag-pipeline-utils/blog/tags/blog/tags/release"},{"inline":false,"label":"Changelog","permalink":"/rag-pipeline-utils/blog/tags/blog/tags/changelog"}],"readingTime":0.15,"hasTruncateMarker":false,"authors":[{"name":"Ali Kahwaji","title":"Creator & Maintainer of RAG Pipeline Utils","url":"https://github.com/alikahwaji","imageURL":"/rag-pipeline-utils/img/authors/ali.jpg","key":"ali","page":null}],"frontMatter":{"slug":"release-v2.1.1","title":"\u2728 Version v2.1.1 Released","authors":["ali"],"tags":["release","changelog"]},"unlisted":false,"prevItem":{"title":"RAG Architecture Overview","permalink":"/rag-pipeline-utils/blog/architecture-overview"}},"content":"RAG Pipeline Utils vv2.1.1 is now available on NPM!\\n\\n#### Highlights\\n- \ud83d\ude80 Auto-published blog from GitHub release\\n- \ud83d\udce6 Version bump: v2.1.1\\n- \ud83d\udcd8 See CHANGELOG.md for full details"}]}}')}}]);