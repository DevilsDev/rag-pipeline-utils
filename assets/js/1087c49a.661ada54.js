"use strict";(self.webpackChunk_devilsdev_rag_pipeline_utils=self.webpackChunk_devilsdev_rag_pipeline_utils||[]).push([[8356],{2404:i=>{i.exports=JSON.parse('{"authors":[{"name":"Ali Kahwaji","title":"Creator & Maintainer of RAG Pipeline Utils","url":"https://github.com/alikahwaji","imageURL":"/rag-pipeline-utils/img/authors/ali.jpg","key":"ali","page":null,"count":7}]}')}}]);