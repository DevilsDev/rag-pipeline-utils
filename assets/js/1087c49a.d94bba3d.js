"use strict";(self.webpackChunk_devilsdev_rag_pipeline_utils=self.webpackChunk_devilsdev_rag_pipeline_utils||[]).push([[8356],{2404:e=>{e.exports=JSON.parse('{"authors":[{"name":"Ali Kahwaji","title":"Creator of RAG Pipeline Utils","url":"https://github.com/DevilsDev","imageURL":"/rag-pipeline-utils/img/authors/ali.jpg","key":"ali","page":null,"count":3}]}')}}]);