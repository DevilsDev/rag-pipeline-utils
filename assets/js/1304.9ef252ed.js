"use strict";(self.webpackChunk_devilsdev_rag_pipeline_utils=self.webpackChunk_devilsdev_rag_pipeline_utils||[]).push([[1304],{1304:(e,t,i)=>{i.r(t),i.d(t,{default:()=>o});i(2610);var n=i(539),s=i(7555),r=i(5208),a=i(3614),l=i(6394);function o(){var e=(0,n.T)({id:"theme.NotFound.title",message:"Page Not Found"});return(0,l.jsxs)(l.Fragment,{children:[(0,l.jsx)(s.be,{title:e}),(0,l.jsx)(r.A,{children:(0,l.jsx)(a.A,{})})]})}},3614:(e,t,i)=>{i.d(t,{A:()=>l});i(2610);var n=i(330),s=i(539),r=i(2576),a=i(6394);function l(e){var t=e.className;return(0,a.jsx)("main",{className:(0,n.A)("container margin-vert--xl",t),children:(0,a.jsx)("div",{className:"row",children:(0,a.jsxs)("div",{className:"col col--6 col--offset-3",children:[(0,a.jsx)(r.A,{as:"h1",className:"hero__title",children:(0,a.jsx)(s.A,{id:"theme.NotFound.title",description:"The title of the 404 page",children:"Page Not Found"})}),(0,a.jsx)("p",{children:(0,a.jsx)(s.A,{id:"theme.NotFound.p1",description:"The first paragraph of the 404 page",children:"We could not find what you were looking for."})}),(0,a.jsx)("p",{children:(0,a.jsx)(s.A,{id:"theme.NotFound.p2",description:"The 2nd paragraph of the 404 page",children:"Please contact the owner of the site that linked you to the original URL and let them know their link is broken."})})]})})})}}}]);