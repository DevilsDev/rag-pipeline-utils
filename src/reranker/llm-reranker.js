/**
 * Version: 1.0.4
 * Description: LLM reranker using LLM to reorder documents based on relevance
 * Author: devilsdev
 * File: src/reranker/llm-reranker.js
 */

export class LLMReranker {
  /**
   * @param {Object} options
   * @param {{ generate(prompt: string): Promise<string> }} options.llm - An LLM client with a `generate` method
   */
  constructor({ llm }) {
    this.llm = llm;
  }

  /**
   * Rerank documents based on the LLM output
   * @param {string} query
   * @param {Array<{ text: string }>} docs
   * @returns {Promise<Array<{ text: string }>>}
   */
  async rerank(query, docs) {
    const prompt = `Given the query: "${query}", rank the following documents by relevance: \n${docs.map((d, i) => `[${i}]: ${d.text}`).join('\n')}\nReturn an array of indices like [2, 0, 1]`;

    try {
      const json = await this.llm.generate(prompt);
      const indices = JSON.parse(json);
      return indices.map(i => docs[i]).filter(Boolean);
    } catch (err) {
      return docs; // Fallback if LLM response is not parsable
    }
  }
}
