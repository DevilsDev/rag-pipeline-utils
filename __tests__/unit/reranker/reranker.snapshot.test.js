/**
 * File: __tests__/unit/reranker/reranker.snapshot.test.js
 * Version: 1.0.0
 * Description: Captures a snapshot baseline of reranked results
 * Author: @devilsdev
 */
import { LLMReranker } from '../../../src/reranker/llm-reranker.js';

test('LLMReranker output snapshot', async () => {
  const mockLLM = { async generate() { return '[1, 0, 2]'; } };
  const mockDocs = [{ text: 'A' }, { text: 'B' }, { text: 'C' }];
  const reranker = new LLMReranker({ llm: mockLLM });
  const result = await reranker.rerank('What is A?', mockDocs);
  expect(result).toMatchSnapshot();
});