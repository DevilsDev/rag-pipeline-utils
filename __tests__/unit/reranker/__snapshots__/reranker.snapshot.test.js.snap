// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LLMReranker output snapshot 1`] = `
[
  {
    "metadata": {
      "source": "test-b.md",
    },
    "pageContent": "Document B",
  },
  {
    "metadata": {
      "source": "test-a.md",
    },
    "pageContent": "Document A",
  },
  {
    "metadata": {
      "source": "test-c.md",
    },
    "pageContent": "Document C",
  },
]
`;
